src/v/kafka/data/cloud_topic_partition.cc

```diff
@@ -74,6 +108,12 @@ static model::record_batch make_placeholder_batch(
       model::record_batch_type::dl_placeholder, hdr.base_offset);
 
     builder.set_producer_identity(hdr.producer_id, hdr.producer_epoch);
+    if (hdr.attrs.is_control()) {
+        builder.set_control_type();
```

Comment:
whats the data flow for control batches in this case? what creates them?

Reply:
Well maybe that is only for consumer offsets topic

Reply:
Here the batch we're creating just replaces the raft data batch (with different type and payload). I don't think it'll have the control bit set. I just tried to copy all fields here. We should probably just copy the header and modify some fields.

Reply:
ah right this is the produce path, not consume path